File: easy/array/reverse_string/reverse_string.go
package main

import "fmt"

func reverseString(s []string) {
	i, j := 0, len(s)-1
	for i < j {
		s[i], s[j] = s[j], s[i]
		i++
		j--
	}
}

func main() {
	a := []string{"h", "e", "l", "l", "o"}
	reverseString(a)
	fmt.Println("Solution 1 : ", a)

	b := []string{"H", "a", "n", "n", "a", "h"}
	reverseString(b)
	fmt.Println("Solution 2 : ", b)
}

File: easy/array/reverse_string/reverse_string.py
class Solution(object):
    def reverseString(self, s):
        """
        :type s: List[str]
        :rtype: None Do not return anything, modify s in-place instead.
        """
        i, j = 0, len(s)-1
        while i < j:
            s[i], s[j] = s[j], s[i]
            i += 1
            j -= 1
        return


s = Solution()
a = ["h", "e", "l", "l", "o"]
s.reverseString(a)
print("Solution 1 : ", a)

b = ["H", "a", "n", "n", "a", "h"]
s.reverseString(b)
print("Solution 2 : ", b)
